allauth/socialaccount/providers/oauth2/views.py

```diff
@@ -57,11 +62,15 @@ def view(request, *args, **kwargs):
 
     def get_client(self, request, app):
         callback_url = reverse(self.adapter.provider_id + "_callback")
-        protocol = (self.adapter.redirect_uri_protocol
-                    or app_settings.DEFAULT_HTTP_PROTOCOL)
-        callback_url = build_absolute_uri(
-            request, callback_url,
-            protocol=protocol)
+        if app_settings.LOGIN_CALLBACK_PROXY:
+            callback_url = urljoin(app_settings.LOGIN_CALLBACK_PROXY, callback_url)
+            callback_url = '%s/proxy/' % callback_url.rstrip('/')
+        else:
+            protocol = (self.adapter.redirect_uri_protocol
+                        or app_settings.DEFAULT_HTTP_PROTOCOL)
+            callback_url = build_absolute_uri(
+                request, callback_url,
```

Comment:
put all the args on one line or split them apart onto separate lines, one or the other